### ISSUE TITLE
Add `UpgradeTimeoutPath` to `24-host`

### ISSUE THREAD

**Issue by damiannolan** [cosmos/ibc-go#1603]
## Summary

> The timeout path is a public path set by the upgrade initiator to determine when the TRY step should timeout. It stores the timeoutHeight and timeoutTimestamp by which point the counterparty must have progressed to the TRY step. The TRY step will prove the timeout values set by the initiating chain and ensure the timeout has not passed. Or in the case of a timeout, in which case counterparty proves that the timeout has passed on its chain and restores the channel.

Add `UpgradeTimeoutPath` to `24-host` with associated key func.

- `UpgradeTimeoutPath`
- `UpgradeTimeoutKey`

Ref: [24-host/keys.go](https://github.com/cosmos/ibc-go/blob/main/modules/core/24-host/keys.go)

```typescript
function timeoutPath(portIdentifier: Identifier, channelIdentifier: Identifier) Path {
    return "channelUpgrade/ports/{portIdentifier}/channelIdentifier/{channelIdentifier}/upgradeTimeout"
}
```

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by damiannolan** [comment]
Review spec for any updates to key paths

**Comment by chatton** [comment]
The spec has been updated with

```js
function channelUpgradeTimeoutPath(portIdentifier: Identifier, channelIdentifier: Identifier) Path {
    return "channelUpgrades/ports/{portIdentifier}/channels/{channelIdentifier}/upgradeTimeout"
}
```

**Comment by chatton** [comment]
closed by #1821
